src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String TITLE_COLUMN = "title";
+    private static final String PRICE_COLUMN = "price";
+    private static final String INSERT_BOOK_SQL = "INSERT INTO book (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_SQL = "SELECT * FROM book WHERE id = ?";
+    private static final String FIND_ALL_SQL = "SELECT * FROM book";
+    private static final String UPDATE_BOOK_SQL =
+            "UPDATE book SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_SQL = "DELETE FROM book WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(INSERT_BOOK_SQL, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("Inserted book is empty");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(FIND_BY_ID_SQL);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (!resultSet.next()) {
+                throw new RuntimeException("No book with id: " + id);
+            }
+            Book book = new Book(resultSet.getString(TITLE_COLUMN),
+                    resultSet.getObject(PRICE_COLUMN, BigDecimal.class));
+            book.setId(id);
+            return Optional.of(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(FIND_ALL_SQL);
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                Book book = new Book(resultSet.getString(TITLE_COLUMN),
+                        resultSet.getObject(PRICE_COLUMN, BigDecimal.class));
+                book.setId(resultSet.getObject(1, Long.class));
```

Comment:
Again, avoid creating an object using both the constructor and setters. Stick to one approach for consistency. Since you're setting both the title and price, consider using a constructor that takes both arguments, or use setters for all properties.